File: messageGo/grpc/baseData/clientMsgData.go
package base_data

// 对应 proto.Vector3
type MultiClientMsgData struct {
	UserList []int64 `json:"userList"` //
	MsgId    int32   `json:"msgId"`
	MsgBody  []byte  `json:"msgBody"` // proto message marshal bytes
}

func (p *MultiClientMsgData) Clear() {
	p.MsgId = 0
	if p.MsgBody != nil {
		p.MsgBody = p.MsgBody[:0]
	}
	if p.UserList != nil {
		p.UserList = p.UserList[:0]
	}
}

File: messageDotNet/grpc/baseData/ClientMsgData.cs
using System;
using UnityEngine;

// 对应 proto.Vector3
[Serializable]
public class MultiClientMsgData : ICustomReference
{
    public long[] UserList;
    public int MsgId;
    public byte[] MsgBody;

    public void Dispose()
    {
        CustomReferencePool.Release(this);
    }
    public static MultiClientMsgData Create(long[] userList, int msgId, byte[] msgBody)
    {
        MultiClientMsgData reference = CustomReferencePool.Acquire<MultiClientMsgData>();
        reference.UserList = userList;
        reference.MsgId = msgId;
        reference.MsgBody = msgBody;
        return reference;
    }
    public void Clear()
    {
        UserList = null;
        MsgId = 0;
        MsgBody = null;
    }
}